src/Resources/config/query_input_visitors.yml

```diff
@@ -0,0 +1,32 @@
+services:
+    EzSystems\EzPlatformGraphQL\GraphQL\InputMapper\Search\Criterion\ContentTypeIdentifier:
+        tags:
+            - {name: 'ezplatform_graphql.query_input_visitor', inputKey: 'ContentTypeIdentifier'}
```

Comment:
We could think of basing the definition of the `ContentSearchQuery` on the defined `inputKey`.

What it implies:
- add a `DomainIterator` that iterates over registered criterion input visitors
- add a worker that adds a field to `ContentSearchQuery` for each of them

The only remaining question is where we get the type of each field from.

Reply:
In this regard, we should keep in mind that we would ideally need an input-object (like `ContentSearchQuery`) that does NOT have Content Type based fields: when filtering on a given content-type, they can't be used (well, they can, but then we will search for content that is of two types, and we don't have that :)